core-data/utils.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * External dependencies
+ */
+import { isPlainObject } from 'lodash';
+
+/**
+ * Module constants
+ */
+const DEFAULT_ID = '[[default]]';
+
+/**
+ * Generates a unique identifier based on an object/scalar query arg.
+ *
+ * @param  {*}      query
+ *
+ * @return {string}       Request ID.
+ */
+export function getRequestId( query = DEFAULT_ID ) {
+	return isPlainObject( query ) ? JSON.stringify( query ) : query;
```

Comment:
`JSON.stringify` is not a trivial operation, particularly for selectors which are called very frequently.

Reply:
What alternative can we use?